sklearn/neighbors/_distance_metric.py

```diff
@@ -0,0 +1,25 @@
+# TODO: Remove this file in 1.2
+import warnings
+
+from ..metrics import DistanceMetric as _DistanceMetric
+
+
+class DistanceMetric(_DistanceMetric):
+    @classmethod
+    def _warn(cls):
+        warnings.warn(
+            "sklearn.neighbors.DistanceMetric has been moved "
+            "to sklearn.metrics.DistanceMetric in 1.0. "
```

Comment:
I hesitated to make it private because this might break other people's code. Thus it is as of now still public via `sklearn.metrics.DistanceMetric`.

Should we make it private?

Reply:
Indeed, it seems that there are legitimate use of this public API...

https://github.com/search?q=%22from+sklearn.neighbors+import+DistanceMetric%22&type=code

So we might want to keep it public. However it would make better sense to move it under `sklearn.metrics.pairwise` instead of directly under`sklearn.metrics`.

Reply:
I think this deprecation can be made in a separate PR to ease the reviews.